test/unit/specs/components/TermsOfUsePage.spec.js

```diff
@@ -29,22 +34,15 @@ describe('TermsOfUsePage.vue', () => {
     expect(wrapper.find({ ref: 'terms-no-button' }).exists()).toBe(true);
   });
 
-  test('clicking Yes commits an acceptTermsOfUse mutation', () => {
-    const mocks = {
-      $router: {
-        push: jest.fn()
-      }
-    };
+  test('clicking Yes changes the route', () => {
     const wrapper = shallow(TermsOfUsePage, { store, localVue, mocks });
-
-    wrapper.vm.acceptTermsOfUse();
-    expect(mutations.acceptTermsOfUse).toHaveBeenCalledWith(state, true);
-    expect(mocks.$router.push).toHaveBeenCalledWith('/');
+    wrapper.find('input[value="Yes"]').trigger('click');
+    expect(mocks.$router.push).toHaveBeenCalledWith('/search');
   });
 
   test('clicking Yes triggers acceptTermsOfUse', () => {
-    const wrapper = shallow(TermsOfUsePage, { store, localVue });
+    const wrapper = shallow(TermsOfUsePage, { store, localVue, mocks });
     wrapper.find('input[value="Yes"]').trigger('click');
-    expect(mutations.acceptTermsOfUse).toHaveBeenCalledWith(state, true);
+    expect(mutations.acceptTermsOfUse).toHaveBeenCalledWith(state, undefined);
   });
 });
```

Comment:
It might be good to have a test for clicking "No" on terms of use. I'm not sure what the behavior is, but from the changes you made it looks like it used to persist the answer either way, and you switched to only saving a "yes" answer? I'm also confused by line 46 above. Why do you need the second parameter if you removed the boolean in that function? expect(mutations.acceptTermsOfUse).toHaveBeenCalledWith(state, undefined);

Reply:
The only change that can be performed on the state is accepting the terms. The "No" button is just a link outside of the site, and the data store is left unchanged. I thought that passing a flag was misleading, particularly given that the function is named `acceptTermsOfUse`. 

The reason for the second parameter in the expectation is that the data store always passes a payload object, even if your handler has an arity of 0.

If the change above is also confusing, I can just rename the mutation to `setTermsOfUseFlag`.

Reply:
I agree that the flag was misleading given the name and that it's better without it. It just wasn't clear from this review whether the "No" had any action that should be testable. If it doesn't, then it's fine as is.